src/app/pathfinding.js

```diff
@@ -0,0 +1,448 @@
+        });
+        return neighbors;
+    }
+    
+    function buildPath(current) {
+        console.log('%c Found a path!', 'color: green');
+        pathNodeArr = [];
+        pathPosArr = [];
+        addNodeToPath(current);
+        
+        computePathLength();
+        // addPathMeshToScene();
+        sendPathToMeasureTool();
+    }
+    
+    let pathNodeArr = [];
+    let pathPosArr = [];
+    let pathLength = null;
+    function addNodeToPath(n) {
+        if (n.parent !== undefined) {
+            pathNodeArr.push(n.parent);
+            // pathPosArr.push(ijToThreejsContainerObjPosition(n.parent.i, n.parent.j));
+            // changeMeshColorFromIndex(ijToIndex(n.parent.i, n.parent.j), GREEN);
+            addNodeToPath(n.parent);
+        }
+    }
+    
+    function _addPathMeshToScene() {
+        const line = new MeshLine();
+        line.setPoints(pathPosArr);
+        const material = new MeshLineMaterial({color: new THREE.Color(0xffff00),lineWidth: 15});
+        const mesh = new THREE.Mesh(line, material);
+        realityEditor.gui.threejsScene.addToScene(mesh);
+    }
+    
+    function sendPathToMeasureTool() {
+        let focusedEnvelopes = realityEditor.envelopeManager.getFocusedEnvelopes();
+        let objectkey = focusedEnvelopes[0].object;
+        let framekey = focusedEnvelopes[0].frame;
+        let gltfBoundingBox = realityEditor.gui.threejsScene.getGltfBoundingBox();
+
+        let groundPlaneMatrix = realityEditor.sceneGraph.getGroundPlaneNode().worldMatrix;
+        let inverseGroundPlaneMatrix = new THREE.Matrix4();
+        realityEditor.gui.threejsScene.setMatrixFromArray(inverseGroundPlaneMatrix, groundPlaneMatrix);
+        inverseGroundPlaneMatrix.invert();
+        
+        if (realityEditor.envelopeManager.getFrameTypeFromKey(objectkey, framekey) === 'spatialMeasure') {
+            let iframe = document.getElementById('iframe' + framekey);
+            iframe.contentWindow.postMessage(JSON.stringify({
+                cellSize: size,
+                pathArr: pathNodeArr,
+                pathLength: pathLength,
+                offset: {
+                    inverseGroundPlaneMatrix: inverseGroundPlaneMatrix,
+                    minX: gltfBoundingBox.min.x * 1000,
+                    minZ: gltfBoundingBox.min.z * 1000,
+                }
+            }), '*');
+        }
+    }
+    
+    function computePathLength() {
+        pathLength = pathNodeArr[pathNodeArr.length - 1].fCost / 10 * size;
+        console.log(`Length of the path is: ${pathLength}`);
+    }
+    
+    function findPath() { // start & end are objects, containing following fields: start.i, start.j, start.index
+        if (Object.keys(start).length === 0 || Object.keys(end).length === 0) {
+            console.warn('Start / end point missing. Need to define both to find path');
+            return;
+        }
+        // console.log(steepnessMapData);
+        // return new Promise((resolve, reject) => {
```

Comment:
Yea I attempted to do a version where findPath() uses async promise to prevent freezing up the main thread when computing path but failed😆I'm building a proper next version with web worker now.

Reply:
Gotcha – webworker version sounds ideal 💯 - though I imagine that could be a significant amount of work – you can leave this here in the meantime if you'd like